### DISCUSSION TITLE
Too easy to accidentally move tablet area in settings

### DISCUSSION THREAD

**Discussion by catsstuffs** [ppy/osu#26539] [Ideas]
### Type

Game behaviour

### Bug description

when using your mouse and dragging to scroll the options menu, its extremely easy to accidentally move the tablet area with no way to undo it besides putting the numbers back in yourself. i've done this personally a few times, would be fine with an undo button or such.

### Screenshots or videos

https://github.com/ppy/osu/assets/123040037/3694b3a5-a609-4d5d-ba0e-37292f58375d

### Version

2023.1224.0-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/13777518/compressed-logs.zip)

**Comment by coltonisgod234** [comment]
### yea this should be fixed... 
i think all the devs need to do is add some code to make it so if you drag the tablet area without holding **SHIFT** it won't do anything. so if you hold shift and move the area around only then it moves the area, and if you drag without holding shift it does nothing? i'm not a game designer, my osu! profile literally says "Unintelligent Person" on it

this is edited, there were mispellings in my last messages

**Comment by LittleEndu** [comment]
Editing the tablet area needs to work in scenarios where you do not have a keyboard attached. Pretty sure you can plug your tablet into your phone and it will work with lazer, but you wouldn't be able to edit the tablet area if it required holding down <kbd>Shift</kbd>

**Comment by marshmallowsunshinekitten1234** [comment]
> Editing the tablet area needs to work in scenarios where you do not have a keyboard attached. Pretty sure you can plug your tablet into your phone and it will work with lazer, but you wouldn't be able to edit the tablet area if it required holding down Shift

Maybe you would want to move the area by using the sliders exclusively in that case since that doesn't require the keyboard key? You're right though, I haven't thought of that actually.

**Comment by mysteGH** [comment]
> Maybe you would want to move the area by using the sliders exclusively in that case since that doesn't require the keyboard key? You're right though, I haven't thought of that actually.

could maybe address this by making it so that you can't change the area whilst the scrollbar is currently moving.
something something if Scroll.InProgress, input.DragToSizeTablet = false (i dont know languages xd)

**Comment by peppy** [comment]
> could maybe address this by making it so that you can't change the area whilst the scrollbar is currently moving.

sounds like a valid path.

**Comment by coltonisgod234** [comment]
> Editing the tablet area needs to work in scenarios where you do not have a keyboard attached. Pretty sure you can plug your tablet into your phone and it will work with lazer, but you wouldn't be able to edit the tablet area if it required holding down <kbd>Shift</kbd> 

Solution: if a keyboard isn't attached disable the functionality entirely, accompany this with a setting.

**Comment by frenzibyte** [comment]
See: https://github.com/ppy/osu/pull/26315#issuecomment-1874832069
